plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/StandardColumnMappings.java

```diff
@@ -327,15 +329,24 @@ public static SliceWriteFunction varbinaryWriteFunction()
         return (statement, index, value) -> statement.setBytes(index, value.getBytes());
     }
 
-    public static ColumnMapping dateColumnMapping()
+    /**
+     * @deprecated This method leads to incorrect result when the date value is before 1582 Oct 14.
+     * If driver supports {@link LocalDate}, use {@link #dateColumnMappingUsingLocalDate} instead.
+     */
+    @Deprecated
+    public static ColumnMapping dateColumnMappingUsingSqlDate()
     {
         return ColumnMapping.longMapping(
                 DATE,
-                dateReadFunction(),
-                dateWriteFunction());
+                dateReadFunctionUsingSqlDate(),
+                dateWriteFunctionUsingSqlDate());
     }
 
-    public static LongReadFunction dateReadFunction()
+    /**
+     * If driver supports {@link LocalDate}, use {@link #dateReadFunctionUsingLocalDate} instead.
```

Comment:
Use `@deprecated` tag for such JavaDoc, shows up well in the IDE (see https://docs.oracle.com/javase/7/docs/technotes/guides/javadoc/deprecation/deprecation.html for full example).